// clear-css/workshop/common/models/book.js
module.exports = function(Book) {
  Book.findOnAmozon = function (ISBN, callback) {
    return callback(null, "Blabla Blabla");
  }
  Book.remoteMethod(
    'findOnAmozon', {
      description: ['arandi'],
      accepts: [{
        arg: 'ISBN',
        type: 'string',
        required: true
      }],
      returns: {
        arg: 'status',
        type: 'string',
        root: true,
        description: 'donen deger'
      },
      https: {
        verb: 'post',
        path: '/findOnAmozon'
      }
    }
  );

  // POST /Books
  // Book.sharedClass.find("create", true).shared = false;

  // GET /Books
  // Book.sharedClass.find("find", true).shared = false;

  // PUT /Books
  Book.sharedClass.find("upsert", true).shared = false;

  // PUT /Books/{id}
  Book.sharedClass.find("updateAttributes", false).shared = false;

  // HEAD /Books/{id}
  Book.sharedClass.find("exists", true).shared = false;

  // GET /Books/{id}
  Book.sharedClass.find("findById", true).shared = false;

  // DELETE /Books/{id}
  Book.sharedClass.find("deleteById", true).shared = false;

  // GET /Books/{id}/exists  Same as HEAD /Books/{id} above
  // Book.sharedClass.find("exists", true).shared = false;

  // GET /Books/count
  Book.sharedClass.find("count", true).shared = false;

  // GET /Books/findOne
  Book.sharedClass.find("findOne", true).shared = false;

  // POST /Books/update updateAll or update use
  Book.sharedClass.find("updateAll", true).shared = false;
};
